src/services/SearchResults.cpp

```diff
@@ -163,7 +168,7 @@ bool SearchResults::ContainsAddress(unsigned int nAddress) const
     return false;
 }
 
-void SearchResults::AddMatches(unsigned int nAddressBase, const unsigned char pMemory[], const std::vector<unsigned int>& vMatches)
+void SearchResults::AddMatches(unsigned int nAddressBase, const unsigned char* restrict pMemory, const std::vector<unsigned int>& vMatches)
```

Comment:
pMemory will always decay into a pointer, there are other places where stuff like this was happend. Plus you can't restrict qualify a C Array.